dds/idl/marshal_generator.cpp

```diff
@@ -2973,7 +2973,12 @@ bool marshal_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
   const ExtensibilityKind exten = be_global->extensibility(node);
   const OpenDDS::DataRepresentation repr =
     be_global->data_representations(node);
+
+  const bool xcdr = repr.xcdr1 || repr.xcdr2;
   const bool not_final = exten != extensibilitykind_final;
+  const bool may_be_parameter_list = exten == extensibilitykind_mutable && xcdr;
+  const bool may_be_delimited = not_final && repr.xcdr2;
+  const bool not_only_delimited = not_final && repr.not_only_xcdr2();
```

Comment:
I know this isn't how the code is now, but we're trying to get rid of generating code based on `repr` for a couple of reasons.